gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -137,11 +137,22 @@
               </div>
             </div>
           </div>
-          <div class="row form-group">
-            <div class="col-md-12">
+          <div class="row">
+            <div class="col-md-6" ng-class="{'has-error': deviceEditor.numConnections.$invalid && deviceEditor.numConnections.$dirty}">
+              <label translate>Connection Management</label>
+              <div class="row">
+                <span class="col-md-8" translate>Number of Connections</span>
+                <div class="col-md-4">
+                  <input name="numConnections" id="numConnections" class="form-control" type="number" pattern="\d+" ng-model="currentDevice.numConnections" min="1" />
```

Comment:
Not really familiar with the rest of the code, but maybe allow the value of `0` to imply the default (i.e. `3` at the moment)? This is currently the case with many other settings (although not all of them, to be exact).

Reply:
The default gets baked when loading so that wouldn't help in this case. I think our handling of defaults is inconsistent and bad and it's on my list for future improvements, but it's tricky...